### PR TITLE
Fix missing -ObjC for XCFrameworks

### PR DESCRIPTION
**Pull request by paulb777**

### CODE REVIEW

**Review by dnkoutso — APPROVED** [review]

File: lib/cocoapods/target/build_settings.rb [hunk 1]
```diff
@@ -1263,8 +1263,24 @@ def other_swift_flags_without_swift?
           end
         end
 
+        # @return [Boolean]
+        define_build_settings_method :any_vendored_static_xcframeworks?, :memoized => true do
+          pod_targets.any? do |pt|
+            pt.build_settings.any? do |bs|
+              if bs.respond_to?(:vendored_xcframeworks)
+                bs.vendored_xcframeworks.any? do |xcf|
```
[the remark this answers is not in the document]
Comment (paulb777) [reply lib/cocoapods/target/build_settings.rb:1271]: It doesn't look like it. See the recent related conversation in #hivemind.

[the remark this answers is not in the document]
Comment (amorde) [reply lib/cocoapods/target/build_settings.rb:1271]: it does but not with the right amount of info. The one referenced here has the "parsed" xcframework

```
# @return [Array<Xcode::XCFramework>]
define_build_settings_method :vendored_xcframeworks, :memoized => true do
  file_accessors.flat_map(&:vendored_xcframeworks).map { |path| load_xcframework(path) }
end
```

[the remark this answers is not in the document]
Comment (amorde) [reply lib/cocoapods/target/build_settings.rb:1271]: won't all `pod_targets` respond to `:vendored_xcframeworks`? wondering if the `respond_to?` check is needed

**Review by amorde — APPROVED** [review]

File: lib/cocoapods/target/build_settings.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulb777) [reply lib/cocoapods/target/build_settings.rb:1271]: `pod_targets`s do, but not `aggregate_targets`. Without the `respond_to?`, there are several unit test failures like:

```
NoMethodError: undefined method `vendored_xcframeworks' for #<Array:0x00007f9e3e0542e8>
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1270:in `block (3 levels) in <class:AggregateTargetSettings>': serializing and deserializing - saves the xcconfig
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1269:in `any?'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1269:in `block (2 levels) in <class:AggregateTargetSettings>'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1268:in `any?'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1268:in `block in <class:AggregateTargetSettings>'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:120:in `block in define_build_settings_method'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1279:in `block in <class:AggregateTargetSettings>'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:120:in `block in define_build_settings_method'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:1292:in `block in <class:AggregateTargetSettings>'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:120:in `block in define_build_settings_method'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:367:in `block in <class:BuildSettings>'
	/Users/paulbeusterien/CocoaPods/CocoaPods/lib/cocoapods/target/build_settings.rb:120:in `block in define_build_settings_method'
```

**Comment by kirualex** [comment]
hopefully this get merged soon, as it will should fix some issues with M1 mac users

File: lib/cocoapods/target/build_settings.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnkoutso) [reply lib/cocoapods/target/build_settings.rb:1271]: weird that this is invoked in aggregate settings, wonder if the test is setup wrong?

**Pull request merged**